### PR TITLE
IGNITE-14110 Module ignite-network created with simple implementation…

### PR DESCRIPTION
**Pull request by akalash**
… based on ScaleCube

### CODE REVIEW

File: modules/network/pom.xml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                <directory>src/main/resources</directory>
+                <filtering>true</filtering>
+            </resource>
+        </resources>
+
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
```
Comment (SammyVimes) [inline modules/network/pom.xml:95]: Compiler plugin is defined and configured in parent pom

**Review by SammyVimes — APPROVED** [review]

**Comment by sergey-chugunov-1985** [comment]
I went through the code, it looks good as a starting point from my perspective. Some abstractions like NetworkCluster are questionable and may need clarification but as a whole piece code looks good.

File: modules/network/src/main/java/org/apache/ignite/network/MessageHandlerHolder.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+     * @param handler Handler for processing all cluster events.
+     */
+    public void addClusterEventHandlers(NetworkClusterEventHandler handler) {
+        clusterEventHandlers.add(handler);
+    }
+
+    /**
+     * @return All handlers for processing incoming messages.
+     */
+    public Collection<NetworkMessageHandler> messageHandlers() {
```
Comment (ygerzhedovich) [inline modules/network/src/main/java/org/apache/ignite/network/MessageHandlerHolder.java:49]: should we return unmodifiable collection here?

File: modules/network/src/main/java/org/apache/ignite/network/MessageHandlerHolder.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+     * @param handler Handler for processing incoming messages.
+     */
+    public void addmessageHandlers(NetworkMessageHandler handler) {
+        messageHandlers.add(handler);
+    }
+
+    /**
+     * @param handler Handler for processing all cluster events.
+     */
+    public void addClusterEventHandlers(NetworkClusterEventHandler handler) {
```
Comment (ygerzhedovich) [inline modules/network/src/main/java/org/apache/ignite/network/MessageHandlerHolder.java:42]: we have 'add' method, but don't have 'remove'

**Comment by ibessonov** [comment]
Code is already merged. All the comments will be addresses in other PRs.

**Pull request closed without merging**